frontend/packages/design-system/src/components/Icon.tsx

```diff
@@ -1,163 +1,31 @@
-      break;
-    case IconName.FAVORITE:
-      iconComponent = icons.Favorite;
-      break;
-    case IconName.FEMALE:
-      iconComponent = icons.Female;
-      break;
-    case IconName.FILTER:
-      iconComponent = icons.Filter;
-      break;
-    case IconName.HOME:
-      iconComponent = icons.Home;
-      break;
-    case IconName.IMAGE:
-      iconComponent = icons.Image;
-      break;
-    case IconName.LOCK:
-      iconComponent = icons.Lock;
-      break;
-    case IconName.MALE:
-      iconComponent = icons.Male;
-      break;
-    case IconName.MENU:
-      iconComponent = icons.Menu;
-      break;
-    case IconName.MORE:
-      iconComponent = icons.More;
-      break;
-    case IconName.NEXT:
-      iconComponent = icons.Next;
-      break;
-    case IconName.NOTIFICATION:
-      iconComponent = icons.Notification;
-      break;
-    case IconName.PLUS:
-      iconComponent = icons.Plus;
-      break;
-    case IconName.PRINT:
-      iconComponent = icons.Print;
-      break;
-    case IconName.PROTECTION:
-      iconComponent = icons.Protection;
-      break;
-    case IconName.REGISTRATION:
-      iconComponent = icons.Registration;
-      break;
-    case IconName.RELOAD:
-      iconComponent = icons.Reload;
-      break;
-    case IconName.SAVE:
-      iconComponent = icons.Save;
-      break;
-    case IconName.SEARCH:
-      iconComponent = icons.Search;
-      break;
-    case IconName.SETTING:
-      iconComponent = icons.Setting;
-      break;
-    case IconName.SHARE:
-      iconComponent = icons.Share;
-      break;
-    case IconName.SYNC:
-      iconComponent = icons.Sync;
-      break;
-    case IconName.THUMBNAIL:
-      iconComponent = icons.Thumbnail;
-      break;
-    case IconName.UNLOCK:
-      iconComponent = icons.Unlock;
-      break;
-    case IconName.UPLOAD:
-      iconComponent = icons.Upload;
-      break;
-    case IconName.USER_ADD:
-      iconComponent = icons.UserAdd;
-      break;
-    case IconName.USER_DELETE:
-      iconComponent = icons.UserDelete;
-      break;
-    case IconName.USER_GROUP:
-      iconComponent = icons.UserGroup;
-      break;
-    case IconName.USER_GROUP_ADD:
-      iconComponent = icons.UserGroupAdd;
-      break;
-    case IconName.WHATSAPP:
-      iconComponent = icons.Whatsapp;
-      break;
-    case IconName.ACTIVITY_CAMP:
-      iconComponent = icons.ActivityCamp;
-      break;
-    case IconName.ACTIVITY_EDUCATION:
-      iconComponent = icons.ActivityEducation;
-      break;
-    case IconName.ACTIVITY_FOOD:
-      iconComponent = icons.ActivityFood;
-      break;
-    case IconName.ACTIVITY_ICLA:
-      iconComponent = icons.ActivityIcla;
-      break;
-    case IconName.ACTIVITY_SHELTER:
-      iconComponent = icons.ActivityShelter;
-      break;
-    case IconName.ACTIVITY_WASH:
-      iconComponent = icons.ActivityWash;
-      break;
-    default:
-      iconComponent = () => {
-        return <Path />;
-      };
-  }
+const Icon: FC<Props> = (props) => {
```

Comment:
Props can be destructed here. I don't think we want to include `name` in the props we send to the native base icon.

`const Icon: FC<Props> = ({ name, ...props }) => {`

Reply:
got an error about rest properties from typescript that way

Reply:
I'm also not getting an error for this. What was it?